system/portenforce.py

```diff
@@ -0,0 +1,182 @@
+      type: string
+      sample: 'tcp'
+    stime:
+      description: The start time of the killed process.
+      type: string
+      sample: Thu Apr 21 17:30:16 2016
+'''
+
+import re
+from subprocess import Popen, PIPE
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[^ ]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('[^ ]+:([^ ]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                proto = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                proto = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('([0-9]+)/(.*)', pidstr)
+            if conns and pids:
+                port = conns.group(1)
+                pid = pids.group(1)
+                name = pids.group(2)
+                result = dict(pid=int(pid), port=int(port), proto=proto, name=name)
+                if result not in results:
+                    results.append(result)
+            elif not pids:
+                raise EnvironmentError('Could not get the pids for the listening ports - possibly a permission issue')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    kill_pids = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            if dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime']) not in kill_pids:
+                kill_pids.append(dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime']))
+    return kill_pids
+
+def getPidSTime(pid):
+    p1 = Popen(['ps', '-o', 'lstart', '-p', str(pid)], stdout=PIPE, stderr=PIPE)
+    p2 = Popen(['grep', '^[^ ]'], stdin=p1.stdout, stdout=PIPE, stderr=PIPE)
+    p1.stdout.close()
+    p3 = Popen(['tr', '-d', '"\n"'], stdin=p2.stdout, stdout=PIPE, stderr=PIPE)
+    p2.stdout.close()
+    stime = p3.communicate()[0]
+    return stime
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=True
+    )
+
+    result = {}
+    result['changed'] = False
+    result['killed'] = list()
+
+    try:
+        # which TCP ports are listening for connections?
+        p1 = Popen(['netstat', '-plnt'], stdout=PIPE, stderr=PIPE)
+        output_tcp = p1.communicate()[0]
+        kill_tcp = netStatParse(output_tcp)
+        for i, p in enumerate(kill_tcp):
+            p['stime'] = getPidSTime(p['pid']) 
+            kill_tcp[i] = p
+
+        # which UDP ports are listening for connections?
+        p1 = Popen(['netstat', '-plnu'], stdout=PIPE, stderr=PIPE)

```

Comment:
If you run a command, you have to use module.get_bin_path, since it will verify the binary is here ( check the others modules for example ). Some distribution no longer install netstat by default (for example, RHEL 7, because net-tools is replaced by iputils)